### ISSUE TITLE
Feature: Custom Layout

### ISSUE THREAD

**Issue by DanCraft99** [desktop/desktop#1674]
The fixed layout of widgets, change/history -> file list -> diff, are laid from left to right.  A flexible layout can fit well to different screen sizes as well as tailored preferences. 

For example, the layout of old GitX has the all the critical elements of git at first launch. 
a) With the change history on top, you can quickly scroll through the changes list.
Note: the same speed as the current placement at the left. (No complaint here)
b) All the file list and changes history below.
You do not need to click file by file to view the changes per file.
Note: this is a feature request.
c) the changes/commit message are at the bottom, where the changes can be viewed and committed in quick seconds.

In summary, it is good to provide custom layout and tailor the needs of different group of developers.

Regards,
Dan

**Comment by joshaber** [comment]
Thanks for the suggestion!

This idea is interesting for the future, but this is beyond the scope of our [current roadmap](https://github.com/desktop/desktop/blob/master/docs/process/roadmap.md).

I'm going to add the `future-proposal` label to this issue and close it out, so that we don't get overwhelmed with our backlog! We'll revisit these as part of our next planning session.

**Comment by billygriffin** [comment]
Thanks for the suggestion @DanCraft99. We think the value of a consistent and expected interface is more important than making everything customizable. We completely understand that developers have differing preferred workflows, and we don't think that GitHub Desktop is going to be the right tool for everyone, but our goal is to build it to be as simple and intuitive as possible to both get up to speed and then to improve the day-to-day workflow of software developers. I'm going to remove `future-proposal` here to indicate that it's not something we intend to support. Thanks again!
